### DISCUSSION TITLE
Run winget on clean install.

### DISCUSSION THREAD

**Discussion by mddtest** [microsoft/winget-cli#1168] [Q&A]
Will it be possible to create a image of Windows 10 now and include a JSON script to get all the programs?

**Comment by denelon** [comment]
You can use `winget export packages.json` to generate a JSON file for all of the programs installed on the machine (assuming a matching manifest is found).

The packages.json file can be used on the "clean" Windows install to import the packages via `winget import packages.json`.

You can also edit the file to identify specific packages you want. The export command also has an argument you can pass to include the versions.

**Reply by mddtest** [reply]
I know of that way already.
I was hoping for a more baked in method for easy installs when needed.
